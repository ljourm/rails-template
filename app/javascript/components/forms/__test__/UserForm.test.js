import { createLocalVue, mount } from "@vue/test-utils"
import Component from "../UserForm"

const email = "test@example.com"
const user = { uuid: "uuid", email: email, name: "name" }

describe("Component", () => {
  const localVue = createLocalVue()

  const wrapper = mount(Component, {
    localVue,
    propsData: { user },
  })

  it("submit event", () => {
    wrapper.find("form").trigger("submit")
    expect(wrapper.emitted("submit")[0][0]).toEqual(user)
  })

  it("destroy event", () => {
    wrapper.find("button.destroy").trigger("click")
    expect(wrapper.emitted("destroy").length).toBe(1)
  })

  it("test", async () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
