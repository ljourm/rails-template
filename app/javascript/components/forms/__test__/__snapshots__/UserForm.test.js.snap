// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test 1`] = `
<form>
  <div
    class="field"
  >
    <label
      class="label"
    >
      氏名
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        name="name"
        type="text"
      />
    </div>
  </div>
  <div
    class="field"
  >
    <label
      class="label"
    >
      EMail
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        name="email"
        type="text"
      />
    </div>
  </div>
  <div
    class="field"
  >
    <label
      class="label"
    >
      パスワード
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        name="password"
        type="password"
      />
    </div>
  </div>
  <div
    class="field"
  >
    <label
      class="label"
    >
      パスワード確認
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        name="password_confirmation"
        type="password"
      />
    </div>
  </div>
  <div
    class="field is-grouped"
  >
    <div
      class="control"
    >
      <button
        class="button is-primary submit"
        type="submit"
      >
        送信
      </button>
    </div>
    <div
      class="control"
    >
      <button
        class="button destroy"
      >
        削除
      </button>
    </div>
  </div>
</form>
`;
